Mathlib/Algebra/AlgebraicCard.lean

```diff
@@ -25,11 +25,11 @@ open Cardinal Polynomial
 
 namespace Algebraic
 
-theorem infinite_of_charZero (R A : Type*) [CommRing R] [IsDomain R] [Ring A] [Algebra R A]
+theorem infinite_of_charZero (R A : Type*) [CommRing R] [Nontrivial R] [Ring A] [Algebra R A]
```

Comment:
If `R = 0`, then `A = 0`, so `CharZero A` is impossible due to `CharZero.infinite`.

Reply:
done, perhaps I should have used the less wordy name `Algebra.nontrivial` instead of `nontrivial_of_nontrivial`?